File: app/src/main/java/com/aresid/simplepasswordgeneratorapp/feature/FeaturesAdapter.java
package com.aresid.simplepasswordgeneratorapp.feature;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.aresid.simplepasswordgeneratorapp.R;

import java.util.ArrayList;

/**
 * Created on: 29/03/2020
 * For Project: pass13
 * Author: René Spies
 * Copyright: © 2020 Ares ID
 */
public class FeaturesAdapter
		extends RecyclerView.Adapter<FeaturesHolder> {
	private static final String             TAG = "FeaturesAdapter";
	private              Context            mContext;
	private              ArrayList<Feature> mFeatures;
	
	public FeaturesAdapter(Context context, ArrayList<Feature> features) {
		Log.d(TAG, "FeaturesAdapter: called");
		mContext = context;
		mFeatures = features;
	}
	
	@NonNull
	@Override
	public FeaturesHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
		Log.d(TAG, "onCreateViewHolder: called");
		View view = LayoutInflater.from(mContext)
		                          .inflate(R.layout.item_unlock_features_recycler_view,
		                                   parent, false);
		return new FeaturesHolder(view);
	}
	
	@Override
	public void onBindViewHolder(@NonNull FeaturesHolder holder, int position) {
		Log.d(TAG, "onBindViewHolder: called");
		Feature feature = mFeatures.get(position);
		holder.setTitle(feature);
	}
	
	@Override
	public int getItemCount() {
		Log.d(TAG, "getItemCount: called");
		return mFeatures.size();
	}
}

File: app/src/main/java/com/aresid/simplepasswordgeneratorapp/feature/FeaturesAdapter.kt
package com.aresid.simplepasswordgeneratorapp.feature

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.aresid.simplepasswordgeneratorapp.R
import java.util.*

/**
 * Created on: 29/03/2020
 * For Project: pass13
 * Author: René Spies
 * Copyright: © 2020 Ares ID
 */
class FeaturesAdapter(
	context: Context,
	features: ArrayList<Feature>
): RecyclerView.Adapter<FeaturesHolder>() {
	
	private val mContext: Context
	private val mFeatures: ArrayList<Feature>
	override fun onCreateViewHolder(
		parent: ViewGroup,
		viewType: Int
	): FeaturesHolder {
		Log.d(
			TAG,
			"onCreateViewHolder: called"
		)
		val view = LayoutInflater.from(mContext).inflate(
			R.layout.item_unlock_features_recycler_view,
			parent,
			false
		)
		return FeaturesHolder(view)
	}
	
	override fun onBindViewHolder(
		holder: FeaturesHolder,
		position: Int
	) {
		Log.d(
			TAG,
			"onBindViewHolder: called"
		)
		val feature = mFeatures[position]
		holder.setTitle(feature)
	}
	
	override fun getItemCount(): Int {
		Log.d(
			TAG,
			"getItemCount: called"
		)
		return mFeatures.size
	}
	
	companion object {
		private const val TAG = "FeaturesAdapter"
	}
	
	init {
		Log.d(
			TAG,
			"FeaturesAdapter: called"
		)
		mContext = context
		mFeatures = features
	}
}